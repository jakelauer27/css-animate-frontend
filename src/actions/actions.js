export const signIn = (user) => ({
  type: 'SIGN_IN',
  user
})

export const signOut = () => ({
  type: 'SIGN_OUT'
})

export const loadPrebuiltAnimations = (prebuiltAnimations) => ({
  type: 'LOAD_PREBUILT_ANIMATIONS',
  prebuiltAnimations
})

export const updateCurrentAnimation = (animation) => ({
  type: 'UPDATE_CURRENT_ANIMATION',
  animation
})

export const saveOriginalAnimation = (animation) => ({
  type: 'SAVE_ORIGINAL_ANIMATION',
  animation
})

export const loadMyAnimations = (myAnimations) => ({
  type: 'LOAD_MY_ANIMATIONS',
  myAnimations
})

export const deleteMyAnimation = (id) => ({
  type: 'DELETE_MY_ANIMATION',
  id
})

export const selectAnimationForEdit = (animation) => ({
  type: 'SELECT_ANIMATION_FOR_EDIT',
  animation
})

export const removeAnimationForEdit = () => ({
  type: 'REMOVE_ANIMATION_FOR_EDIT'
})


