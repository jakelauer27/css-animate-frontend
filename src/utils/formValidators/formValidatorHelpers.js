export const invalidInput = (target) => {
  target.classList.add('red')
}

export const validInput = (target) => {
  target.classList.remove('red')
}
