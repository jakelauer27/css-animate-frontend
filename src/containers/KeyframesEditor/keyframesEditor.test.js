import { KeyframesEditor, mapStateToProps, mapDispatchToProps } from './KeyframesEditor'
import React from 'react'
import { shallow } from 'enzyme'
import { loadAnimation } from '../../actions/actions'
import { updateKeyframes } from '../../utils/keyframesInsertion'
import * as formValidation from '../../utils/formValidators'

jest.mock('../../utils/formValidators')
jest.mock('../../utils/keyframesInsertion')

describe('keyframesEditor', () => {
  let wrapper
  const mockUpdateAnimation = jest.fn()
  const mockAnimation = { 
    'properties': {
      'name': 'slideInX',
      'duration': '1.5s',
      'timingFunction': 'ease',
      'delay': '0s',
      'iterationCount': '1',
      'direction': 'normal',
      'fillMode': 'forwards'
    },
    'keyframes': {
      'name': 'slideInX',
      'sections': [
        {
          'name': '0%',
          'label': '0%',
          'properties': [
            {
              'name': 'transform',
              'value': 'translateX(-300px)'
            }
          ]
        },
        {
          'name': '100%',
          'label': '100%',
          'properties': [
            {
              'name': 'transform',
              'value': 'translateX(0px)'
            }
          ]
        },
      ]
    }
  }
  const mockAltAnimation = { 
    'properties': {
      'name': 'slideInX',
      'duration': '1.5s',
      'timingFunction': 'ease',
      'delay': '0s',
      'iterationCount': '1',
      'direction': 'normal',
      'fillMode': 'forwards'
    },
    'keyframes': {
      'name': 'slideInX',
      'sections': [
        {
          'name': '0%',
          'label': '0%',
          'properties': [
            {
              'name': 'transform',
              'value': 'translateX(-300px)'
            }
          ]
        },
      ]
    }
  }

  beforeEach(() => {
    document.querySelector = jest.fn().mockImplementation(() => ({
      click: jest.fn()
    }))
    wrapper = shallow(<KeyframesEditor animation={mockAnimation} updateAnimation={mockUpdateAnimation} />)
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot if there is no animation', () => {
    wrapper = shallow(<KeyframesEditor animation={{keyframes: null}} updateAnimation={mockUpdateAnimation} currentAnimation={'slideInX'}/>)
    expect(wrapper).toMatchSnapshot()
  })

  it('should call saveKeyframesStages when a label input is changed', () => {
    wrapper = shallow(<KeyframesEditor animation={mockAltAnimation} updateAnimation={mockUpdateAnimation} currentAnimation={'slideInX'}/>)
    const spy = jest.spyOn(wrapper.instance(), 'saveKeyframesStages')
    const mockEvent = {target: {value: '3%', classList: ['input', '0%']}}
    
    wrapper.find('.keyframes-label').simulate('change', mockEvent)
    expect(spy).toHaveBeenCalledWith(mockEvent)
  })

  it('should call saveKeyframesProps when a value input is changed', () => {
    wrapper = shallow(<KeyframesEditor animation={mockAltAnimation} updateAnimation={mockUpdateAnimation} currentAnimation={'slideInX'}/>)
    const spy = jest.spyOn(wrapper.instance(), 'saveKeyframesProps')
    const mockEvent = {target: {value: 'rotateY(30deg)', classList: ['input', 'transform'], parentElement: {parentElement: {childNodes: [{classList: [null, '0%']}]}}}}
    
    wrapper.find('.keyframe-prop-value').simulate('change', mockEvent)
    expect(spy).toHaveBeenCalledWith(mockEvent)
  })

  describe('saveKeyframesStages', () => {
      let mockEvent;

    beforeEach(() => {
      mockEvent = {target: {value: '3%', classList: ['input', '0%']}}
    })
    
    it('should call formvalidation to validate the input', () => {
      const spy = jest.spyOn(formValidation, 'keyframeStage')
      wrapper.instance().saveKeyframesStages(mockEvent)
      expect(spy).toHaveBeenCalledWith(mockEvent.target, mockEvent.target.value)
    })

    it('should click the stop btn on the dom', () => {
      wrapper.instance().saveKeyframesStages(mockEvent)
      expect(document.querySelector).toHaveBeenCalledWith('.stop-btn')
    })

    it('should call updateAnimation passing in an updated animation with a % if the value is empty', () => {
      mockEvent = {target: {value: '', classList: ['input', '0%']}}
      const expectedAnimation = {"keyframes": {"name": "slideInX", "sections": [{"label": "%", "name": "0%", "properties": [{"name": "transform", "value": "translateX(-300px)"}]}, {"label": "100%", "name": "100%", "properties": [{"name": "transform", "value": "translateX(0px)"}]}]}, "properties": {"delay": "0s", "direction": "normal", "duration": "1.5s", "fillMode": "forwards", "iterationCount": "1", "name": "slideInX", "timingFunction": "ease"}}
      wrapper.instance().saveKeyframesStages(mockEvent)
      expect(mockUpdateAnimation).toHaveBeenCalledWith(expectedAnimation)
    })

    it('should call updateAnimation passing in an updated animation with the input value', () => {
      const expectedAnimation = {"keyframes": {"name": "slideInX", "sections": [{"label": "3%", "name": "0%", "properties": [{"name": "transform", "value": "translateX(-300px)"}]}, {"label": "100%", "name": "100%", "properties": [{"name": "transform", "value": "translateX(0px)"}]}]}, "properties": {"delay": "0s", "direction": "normal", "duration": "1.5s", "fillMode": "forwards", "iterationCount": "1", "name": "slideInX", "timingFunction": "ease"}}
      wrapper.instance().saveKeyframesStages(mockEvent)
      expect(mockUpdateAnimation).toHaveBeenCalledWith(expectedAnimation)
    })

    it('should call updateKeyframes passing in an updated keyframes with the input value', () => {
      updateKeyframes.mockImplementation(() => jest.fn())
      const expectedAnimation = {"keyframes": {"name": "slideInX", "sections": [{"label": "3%", "name": "0%", "properties": [{"name": "transform", "value": "translateX(-300px)"}]}, {"label": "100%", "name": "100%", "properties": [{"name": "transform", "value": "translateX(0px)"}]}]}, "properties": {"delay": "0s", "direction": "normal", "duration": "1.5s", "fillMode": "forwards", "iterationCount": "1", "name": "slideInX", "timingFunction": "ease"}}
      wrapper.instance().saveKeyframesStages(mockEvent)
      expect(updateKeyframes).toHaveBeenCalledWith(expectedAnimation.keyframes)
    })
  })

  describe('saveKeyframesProps', () => {
    let mockEvent;

    beforeEach(() => {
      mockEvent = {target: {value: 'rotateY(30deg)', classList: ['input', 'transform'], parentElement: {parentElement: {childNodes: [{classList: [null, '0%']}]}}}}
      document.querySelector = jest.fn().mockImplementation(() => ({
        click: jest.fn()
      }))
    })
    
    it('should call formvalidation to validate the input', () => {
      const spy = jest.spyOn(formValidation, 'keyframeValue')
      wrapper.instance().saveKeyframesProps(mockEvent)
      expect(spy).toHaveBeenCalledWith(mockEvent.target, mockEvent.target.value)
    })

    it('should click the stop btn on the dom', () => {
      wrapper.instance().saveKeyframesProps(mockEvent)
      expect(document.querySelector).toHaveBeenCalledWith('.stop-btn')
    })

    it('should call updateAnimation passing in an updated animation with the input value', () => {
      const expectedAnimation = {"keyframes": {"name": "slideInX", "sections": [{"label": "0%", "name": "0%", "properties": [{"name": "transform", "value": "rotateY(30deg)"}]}, {"label": "100%", "name": "100%", "properties": [{"name": "transform", "value":"translateX(0px)"}]}]}, "properties": {"delay": "0s", "direction": "normal", "duration": "1.5s", "fillMode": "forwards", "iterationCount": "1", "name": "slideInX", "timingFunction": "ease"}}
      wrapper.instance().saveKeyframesProps(mockEvent)
      expect(mockUpdateAnimation).toHaveBeenCalledWith(expectedAnimation)
    })

    it('should call updateKeyframes passing in an updated keyframes with the input value', () => {
      updateKeyframes.mockImplementation(() => jest.fn())
      const expectedAnimation = {"name": "slideInX", "sections": [{"label": "0%", "name": "0%", "properties": [{"name": "transform", "value": "rotateY(30deg)"}]}, {"label": "100%", "name": "100%", "properties": [{"name": "transform", "value": "translateX(0px)"}]}]}
      wrapper.instance().saveKeyframesProps(mockEvent)
      expect(updateKeyframes).toHaveBeenCalledWith(expectedAnimation)
    })

  })

  describe('mapStateToProps', () => {

    it('should return a props object with an animation property', () => {
      const mockState = {animation: mockAnimation}
      const expected = {animation: mockAnimation}
      expect(mapStateToProps(mockState)).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {

    it('should return a props object with a loadNewAnimation property', () => {
        const mockDispatch = jest.fn()
        const actionToDispatch = loadAnimation(mockAnimation)
        const mappedProps = mapDispatchToProps(mockDispatch)
        mappedProps.updateAnimation(mockAnimation)
        expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch)
    })
  })
})
