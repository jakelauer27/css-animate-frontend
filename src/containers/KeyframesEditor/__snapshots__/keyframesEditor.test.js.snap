// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keyframesEditor should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyframesEditor
    animation={
      Object {
        "keyframes": Object {
          "name": "slideInX",
          "sections": Array [
            Object {
              "label": "0%",
              "name": "0%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(-300px)",
                },
              ],
            },
            Object {
              "label": "100%",
              "name": "100%",
              "properties": Array [
                Object {
                  "name": "transform",
                  "value": "translateX(0px)",
                },
              ],
            },
          ],
        },
        "properties": Object {
          "delay": "0s",
          "direction": "normal",
          "duration": "1.5s",
          "fillMode": "forwards",
          "iterationCount": "1",
          "name": "slideInX",
          "timingFunction": "ease",
        },
      }
    }
    updateAnimation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="keyframe"
      >
        <p
          className="keyframes-name"
        >
          <span>
            @keyframes 
          </span>
          slideInX
           
          <span>
             
            {
          </span>
        </p>
        <div
          className="keyframes-section"
        >
          <input
            className="keyframes-label 0%"
            onChange={[Function]}
            type="text"
            value="0%"
          />
          <span>
             
            {
          </span>
          <div
            className="props-container"
          >
            <p
              className="keyframe-property"
            >
              transform
              <span>
                :
              </span>
            </p>
            <input
              className="keyframe-prop-value transform"
              onChange={[Function]}
              type="text"
              value="translateX(-300px)"
            />
          </div>
          <p
            className="close-curly"
          >
            }
          </p>
        </div>
        <div
          className="keyframes-section"
        >
          <input
            className="keyframes-label 100%"
            onChange={[Function]}
            type="text"
            value="100%"
          />
          <span>
             
            {
          </span>
          <div
            className="props-container"
          >
            <p
              className="keyframe-property"
            >
              transform
              <span>
                :
              </span>
            </p>
            <input
              className="keyframe-prop-value transform"
              onChange={[Function]}
              type="text"
              value="translateX(0px)"
            />
          </div>
          <p
            className="close-curly"
          >
            }
          </p>
        </div>
        <p
          className="close-curly"
        >
          }
        </p>
      </div>,
      "className": "keyframes-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="keyframes-name"
          >
            <span>
              @keyframes 
            </span>
            slideInX
             
            <span>
               
              {
            </span>
          </p>,
          Array [
            <div
              className="keyframes-section"
            >
              <input
                className="keyframes-label 0%"
                onChange={[Function]}
                type="text"
                value="0%"
              />
              <span>
                 
                {
              </span>
              <div
                className="props-container"
              >
                <p
                  className="keyframe-property"
                >
                  transform
                  <span>
                    :
                  </span>
                </p>
                <input
                  className="keyframe-prop-value transform"
                  onChange={[Function]}
                  type="text"
                  value="translateX(-300px)"
                />
              </div>
              <p
                className="close-curly"
              >
                }
              </p>
            </div>,
            <div
              className="keyframes-section"
            >
              <input
                className="keyframes-label 100%"
                onChange={[Function]}
                type="text"
                value="100%"
              />
              <span>
                 
                {
              </span>
              <div
                className="props-container"
              >
                <p
                  className="keyframe-property"
                >
                  transform
                  <span>
                    :
                  </span>
                </p>
                <input
                  className="keyframe-prop-value transform"
                  onChange={[Function]}
                  type="text"
                  value="translateX(0px)"
                />
              </div>
              <p
                className="close-curly"
              >
                }
              </p>
            </div>,
          ],
          <p
            className="close-curly"
          >
            }
          </p>,
        ],
        "className": "keyframe",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span>
                @keyframes 
              </span>,
              "slideInX",
              " ",
              <span>
                 
                {
              </span>,
            ],
            "className": "keyframes-name",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "@keyframes ",
              },
              "ref": null,
              "rendered": "@keyframes ",
              "type": "span",
            },
            "slideInX",
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "{",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "{",
              ],
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": "uid1",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="keyframes-label 0%"
                onChange={[Function]}
                type="text"
                value="0%"
              />,
              <span>
                 
                {
              </span>,
              Array [
                <div
                  className="props-container"
                >
                  <p
                    className="keyframe-property"
                  >
                    transform
                    <span>
                      :
                    </span>
                  </p>
                  <input
                    className="keyframe-prop-value transform"
                    onChange={[Function]}
                    type="text"
                    value="translateX(-300px)"
                  />
                </div>,
              ],
              <p
                className="close-curly"
              >
                }
              </p>,
            ],
            "className": "keyframes-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "keyframes-label 0%",
                "onChange": [Function],
                "type": "text",
                "value": "0%",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "{",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "{",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "uid2",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="keyframe-property"
                  >
                    transform
                    <span>
                      :
                    </span>
                  </p>,
                  <input
                    className="keyframe-prop-value transform"
                    onChange={[Function]}
                    type="text"
                    value="translateX(-300px)"
                  />,
                ],
                "className": "props-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "transform",
                      <span>
                        :
                      </span>,
                    ],
                    "className": "keyframe-property",
                  },
                  "ref": null,
                  "rendered": Array [
                    "transform",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": ":",
                      },
                      "ref": null,
                      "rendered": ":",
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "keyframe-prop-value transform",
                    "onChange": [Function],
                    "type": "text",
                    "value": "translateX(-300px)",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "}",
                "className": "close-curly",
              },
              "ref": null,
              "rendered": "}",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "uid3",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="keyframes-label 100%"
                onChange={[Function]}
                type="text"
                value="100%"
              />,
              <span>
                 
                {
              </span>,
              Array [
                <div
                  className="props-container"
                >
                  <p
                    className="keyframe-property"
                  >
                    transform
                    <span>
                      :
                    </span>
                  </p>
                  <input
                    className="keyframe-prop-value transform"
                    onChange={[Function]}
                    type="text"
                    value="translateX(0px)"
                  />
                </div>,
              ],
              <p
                className="close-curly"
              >
                }
              </p>,
            ],
            "className": "keyframes-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "keyframes-label 100%",
                "onChange": [Function],
                "type": "text",
                "value": "100%",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "{",
                ],
              },
              "ref": null,
              "rendered": Array [
                " ",
                "{",
              ],
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "uid4",
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="keyframe-property"
                  >
                    transform
                    <span>
                      :
                    </span>
                  </p>,
                  <input
                    className="keyframe-prop-value transform"
                    onChange={[Function]}
                    type="text"
                    value="translateX(0px)"
                  />,
                ],
                "className": "props-container",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "transform",
                      <span>
                        :
                      </span>,
                    ],
                    "className": "keyframe-property",
                  },
                  "ref": null,
                  "rendered": Array [
                    "transform",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": ":",
                      },
                      "ref": null,
                      "rendered": ":",
                      "type": "span",
                    },
                  ],
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "keyframe-prop-value transform",
                    "onChange": [Function],
                    "type": "text",
                    "value": "translateX(0px)",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "}",
                "className": "close-curly",
              },
              "ref": null,
              "rendered": "}",
              "type": "p",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "}",
            "className": "close-curly",
          },
          "ref": null,
          "rendered": "}",
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="keyframe"
        >
          <p
            className="keyframes-name"
          >
            <span>
              @keyframes 
            </span>
            slideInX
             
            <span>
               
              {
            </span>
          </p>
          <div
            className="keyframes-section"
          >
            <input
              className="keyframes-label 0%"
              onChange={[Function]}
              type="text"
              value="0%"
            />
            <span>
               
              {
            </span>
            <div
              className="props-container"
            >
              <p
                className="keyframe-property"
              >
                transform
                <span>
                  :
                </span>
              </p>
              <input
                className="keyframe-prop-value transform"
                onChange={[Function]}
                type="text"
                value="translateX(-300px)"
              />
            </div>
            <p
              className="close-curly"
            >
              }
            </p>
          </div>
          <div
            className="keyframes-section"
          >
            <input
              className="keyframes-label 100%"
              onChange={[Function]}
              type="text"
              value="100%"
            />
            <span>
               
              {
            </span>
            <div
              className="props-container"
            >
              <p
                className="keyframe-property"
              >
                transform
                <span>
                  :
                </span>
              </p>
              <input
                className="keyframe-prop-value transform"
                onChange={[Function]}
                type="text"
                value="translateX(0px)"
              />
            </div>
            <p
              className="close-curly"
            >
              }
            </p>
          </div>
          <p
            className="close-curly"
          >
            }
          </p>
        </div>,
        "className": "keyframes-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <p
              className="keyframes-name"
            >
              <span>
                @keyframes 
              </span>
              slideInX
               
              <span>
                 
                {
              </span>
            </p>,
            Array [
              <div
                className="keyframes-section"
              >
                <input
                  className="keyframes-label 0%"
                  onChange={[Function]}
                  type="text"
                  value="0%"
                />
                <span>
                   
                  {
                </span>
                <div
                  className="props-container"
                >
                  <p
                    className="keyframe-property"
                  >
                    transform
                    <span>
                      :
                    </span>
                  </p>
                  <input
                    className="keyframe-prop-value transform"
                    onChange={[Function]}
                    type="text"
                    value="translateX(-300px)"
                  />
                </div>
                <p
                  className="close-curly"
                >
                  }
                </p>
              </div>,
              <div
                className="keyframes-section"
              >
                <input
                  className="keyframes-label 100%"
                  onChange={[Function]}
                  type="text"
                  value="100%"
                />
                <span>
                   
                  {
                </span>
                <div
                  className="props-container"
                >
                  <p
                    className="keyframe-property"
                  >
                    transform
                    <span>
                      :
                    </span>
                  </p>
                  <input
                    className="keyframe-prop-value transform"
                    onChange={[Function]}
                    type="text"
                    value="translateX(0px)"
                  />
                </div>
                <p
                  className="close-curly"
                >
                  }
                </p>
              </div>,
            ],
            <p
              className="close-curly"
            >
              }
            </p>,
          ],
          "className": "keyframe",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <span>
                  @keyframes 
                </span>,
                "slideInX",
                " ",
                <span>
                   
                  {
                </span>,
              ],
              "className": "keyframes-name",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "@keyframes ",
                },
                "ref": null,
                "rendered": "@keyframes ",
                "type": "span",
              },
              "slideInX",
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "{",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "{",
                ],
                "type": "span",
              },
            ],
            "type": "p",
          },
          Object {
            "instance": null,
            "key": "uid1",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="keyframes-label 0%"
                  onChange={[Function]}
                  type="text"
                  value="0%"
                />,
                <span>
                   
                  {
                </span>,
                Array [
                  <div
                    className="props-container"
                  >
                    <p
                      className="keyframe-property"
                    >
                      transform
                      <span>
                        :
                      </span>
                    </p>
                    <input
                      className="keyframe-prop-value transform"
                      onChange={[Function]}
                      type="text"
                      value="translateX(-300px)"
                    />
                  </div>,
                ],
                <p
                  className="close-curly"
                >
                  }
                </p>,
              ],
              "className": "keyframes-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "keyframes-label 0%",
                  "onChange": [Function],
                  "type": "text",
                  "value": "0%",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "{",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "{",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "uid2",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="keyframe-property"
                    >
                      transform
                      <span>
                        :
                      </span>
                    </p>,
                    <input
                      className="keyframe-prop-value transform"
                      onChange={[Function]}
                      type="text"
                      value="translateX(-300px)"
                    />,
                  ],
                  "className": "props-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "transform",
                        <span>
                          :
                        </span>,
                      ],
                      "className": "keyframe-property",
                    },
                    "ref": null,
                    "rendered": Array [
                      "transform",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": ":",
                        },
                        "ref": null,
                        "rendered": ":",
                        "type": "span",
                      },
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "keyframe-prop-value transform",
                      "onChange": [Function],
                      "type": "text",
                      "value": "translateX(-300px)",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "}",
                  "className": "close-curly",
                },
                "ref": null,
                "rendered": "}",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "uid3",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  className="keyframes-label 100%"
                  onChange={[Function]}
                  type="text"
                  value="100%"
                />,
                <span>
                   
                  {
                </span>,
                Array [
                  <div
                    className="props-container"
                  >
                    <p
                      className="keyframe-property"
                    >
                      transform
                      <span>
                        :
                      </span>
                    </p>
                    <input
                      className="keyframe-prop-value transform"
                      onChange={[Function]}
                      type="text"
                      value="translateX(0px)"
                    />
                  </div>,
                ],
                <p
                  className="close-curly"
                >
                  }
                </p>,
              ],
              "className": "keyframes-section",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "keyframes-label 100%",
                  "onChange": [Function],
                  "type": "text",
                  "value": "100%",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "{",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "{",
                ],
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "uid4",
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="keyframe-property"
                    >
                      transform
                      <span>
                        :
                      </span>
                    </p>,
                    <input
                      className="keyframe-prop-value transform"
                      onChange={[Function]}
                      type="text"
                      value="translateX(0px)"
                    />,
                  ],
                  "className": "props-container",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        "transform",
                        <span>
                          :
                        </span>,
                      ],
                      "className": "keyframe-property",
                    },
                    "ref": null,
                    "rendered": Array [
                      "transform",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": ":",
                        },
                        "ref": null,
                        "rendered": ":",
                        "type": "span",
                      },
                    ],
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "keyframe-prop-value transform",
                      "onChange": [Function],
                      "type": "text",
                      "value": "translateX(0px)",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "}",
                  "className": "close-curly",
                },
                "ref": null,
                "rendered": "}",
                "type": "p",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "}",
              "className": "close-curly",
            },
            "ref": null,
            "rendered": "}",
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`keyframesEditor should match the snapshot if there is no animation 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <KeyframesEditor
    animation={
      Object {
        "keyframes": null,
      }
    }
    currentAnimation="slideInX"
    updateAnimation={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
